### ISSUE TITLE
md-select possible scope conflict

### ISSUE THREAD

**Issue by tam3r** [angular/material#8004]
There is the following line in select.js:
 `scope.$watch(selectMenuCtrl.selectedLabels, syncLabelText)`

As a result `selectMenuCtrl.selectedLabels(scope)` is executed with every $digest .

Take a look at the selectedLabels definition:

```
self.selectedLabels = function(opts) {
  opts = opts || {};
  var mode = opts.mode || 'html';
  var selectedOptionEls = $mdUtil.nodesToArray($element[0].querySelectorAll('md-option[selected]'));
  if (selectedOptionEls.length) {
    var mapFn;

    if (mode == 'html') {
      // Map the given element to its innerHTML string. If the element has a child ripple
      // container remove it from the HTML string, before returning the string.
      mapFn = function(el) {
        var html = el.innerHTML;
        // Remove the ripple container from the selected option, copying it would cause a CSP violation.
        var rippleContainer = el.querySelector('.md-ripple-container');
        return rippleContainer ? html.replace(rippleContainer.outerHTML, '') : html;
      };
    } else if (mode == 'aria') {
      mapFn = function(el) {
        return el.hasAttribute('aria-label') ? el.getAttribute('aria-label') : el.textContent;
      };
    }
    return selectedOptionEls.map(mapFn).join(', ');
  } else {
    return '';
  }
};
```

The issue is that if you have a variable named "mode" on the scope (not an impossible scenario, obviously))), that leads to either mapFn being `undefined` and therefore an error in the line:

`selectedOptionEls.map(mapFn).join(', ');`

or simply to unwanted behavior.

**Comment by Comandante** [comment]
In addition, If you define variable "mode" in your scope - opts.mode will intercept this variable and mapFn will be undefined again.

**Comment by topherfangio** [comment]
This appears to have been fixed in d0bacdfeff925b3c02338a0fa4c184effdc4e30f.

@tam3r @Comandante Can one of you confirm that the issue is resolved in master?

**Comment by tam3r** [comment]
Yep, seems to be okay. Thanks for great work.
